sync/once_map.go

```diff
@@ -0,0 +1,43 @@
+package sync
+
+import (
+	"sync"
+)
+
+// MutexMap is a map of mutexes where key is a string and value is a sync.RWMutex.
+type OnceMap struct {
+	mu      sync.RWMutex
+	onceMap map[string]*sync.Once
```

Comment:
in general, is can be a map[string]struct{}, just to store the keys, for which we called `f`.
we never use Once stored in that map